### PR TITLE
Portal support for DS records

### LINKED ISSUE #443: Add DS to acl logic
Portal and api
- Add it to the list in the portal when making an ACL rule
- Verify that you can add a DS ACL rule in the api...may not require changes

Related to #421

### PR DESCRIPTION
**Pull request by mitruly**
Adds portal modal and service support for DS records. Fixes #443.

Changes in this pull request:
- Add support for DS records in ACL rule modal
- Add conversion for DS record to display record in record set info

### CODE REVIEW

**Comment by mitruly** [comment]
Portal images with changes for DS records:

image
image
image
image

Note: View record set info was not updated as `SOA` is also not supported.

File: modules/portal/public/lib/controllers/controller.manageZones.js [hunk 1]
```diff
@@ -63,6 +63,7 @@ angular.module('controller.manageZones', [])
             readOnly: false
         }
     };
+    $scope.aclRecordTypes = ['A', 'AAAA', 'CNAME', 'DS', 'MX', 'NS', 'TXT', 'SRV', 'PTR', 'SPF', 'SSHFP'];
```
Comment (mitruly) [inline modules/portal/public/lib/controllers/controller.manageZones.js]: Needed to define a separate list for record types that ACL can be created for, as this differs from the record create/view list.

File: modules/portal/app/views/zones/zoneTabs/manageRecords.scala.html [hunk 2]
```diff
@@ -147,19 +147,27 @@ <h3 class="panel-title">Records {{ getRecordPageTitle() }}</h3>
                                 </li>
                             </ul>
                         </span>
-                        <span ng-if="record.type == 'SPF'">
```
Comment (mitruly) [inline modules/portal/app/views/zones/zoneTabs/manageRecords.scala.html]: Reordered types for readability.

**Comment by britneywright** [comment]
should reorder SRV and SPF record types here in keeping with the alphabetized ordering. https://github.com/vinyldns/vinyldns/blob/de5bf17676df3ad5a42cb3f401f2ea93b28ad994/modules/portal/public/lib/services/records/service.records.js#L153-L166

**Pull request merged**